### PR TITLE
Add typescript classes guide.

### PR DESCRIPTION
**Pull request by jasmeetsohal**
- [X] I have read freeCodeCamp's contribution guidelines.
- [x] My pull request has a descriptive title (not a vague title like Update index.md)
- [x] My pull request targets the master branch of freeCodeCamp.
- [X]  None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

### CODE REVIEW

**Review by jasmeetsohal — COMMENTED** [review]
text formatting changes

File: guide/english/typescript/classes/index.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      }
+```
+Class can embrace fields, constructors and functions in it.
+
+```typescript
+class Pet {  
+   name:string; 
+ 
+   constructor(name:string) { 
+      this.name = name
```
Comment (Manish-Giri) [inline guide/english/typescript/classes/index.md]: Add the ending `;` to be consistent. 
```suggestion
      this.name = name; 
```

File: guide/english/typescript/classes/index.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+```typescript
+class Pet {  
+   name:string; 
+ 
+   constructor(name:string) { 
+      this.name = name 
+   }  
+
+   function show():void { 
+      console.log("Pet name is  :   "+this.name)
```
Comment (Manish-Giri) [inline guide/english/typescript/classes/index.md]: Add ending `;`.
```suggestion
      console.log("Pet name is  :   "+this.name); 
```

File: guide/english/typescript/classes/index.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      console.log("Pet name is  :   "+this.name) 
+   } 
+}
+```
+To access the attributes of class `Pet`, It needs to create an instance of the class like under the hood -
+
+```typescript
+let pet = new Pet('Bruno');
+
+pet.show();
```
Comment (Manish-Giri) [inline guide/english/typescript/classes/index.md]: Add the expected output in a comment beside this line -
```suggestion
pet.show(); // output goes here
```

File: guide/english/typescript/classes/index.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+# Classes
+
+`Object Oriented Programming` is worthwhile where tons of code are written, by structuring the code it makes vivid to understand.
+Class is one of the paradigm of `OOP`.
+
+TypeScript class reserved `class` keyword to create a class. The syntax is given below - 
+
+```typescript
+class <class_name>{
+      //Fields
```
Comment (Manish-Giri) [inline guide/english/typescript/classes/index.md]: Add a space after the comment and the comment text -
```suggestion
      // Fields
      // Constructors
      // Functions
```

File: guide/english/typescript/classes/index.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
+---
+title: Classes
+---
+
+# Classes
+
+`Object Oriented Programming` is worthwhile where tons of code are written, by structuring the code it makes vivid to understand.
```
Comment (Manish-Giri) [inline guide/english/typescript/classes/index.md]: Please re-phrase these lines so they are easier to understand.
> `Object Oriented Programming` is worthwhile where tons of code are written, by structuring the code it makes vivid to understand.
Class is one of the paradigm of `OOP`.
TypeScript class reserved `class` keyword to create a class. The syntax is given below - 

Maybe something like this -

```suggestion
Object Oriented Programming is an important programming principle, especially where a lot of code is written. By structuring the code in classes and objects, it makes the code easier to understand.
```

Same with this line -
> TypeScript class reserved `class` keyword to create a class. The syntax is given below -

**Review by Manish-Giri — CHANGES_REQUESTED** [review]
Hi @jasmeetsohal 

Please re-open this PR if you wish to apply the suggested changes.

**Comment by jasmeetsohal** [comment]
@Manish-Giri Where do I re-open this PR? I couldn't find the way to re-open this.

**Comment by scissorsneedfoodtoo** [comment]
@jasmeetsohal, it should be the button on the left right below this comment box. I can go ahead an reopen it for you.

**Comment by jasmeetsohal** [comment]
Hi @Manish-Giri 
Please review the changes that I've made.

**Comment by jasmeetsohal** [comment]
> @jasmeetsohal, it should be the button on the left right below this comment box. I can go ahead an reopen it for you.

Thanks @scissorsneedfoodtoo

**Review by Manish-Giri — APPROVED** [review]
Hi @jasmeetsohal 

Thanks for making the changes.

While most of your changes are good, I was meaning for you to add the expected output when this line would be executed - 

> `pet.show();`

But I've fixed it and added the output - 

> `pet.show(); // Pet name is: Bruno`

Thanks for creating the PR, and Happy Coding!

**Comment by jasmeetsohal** [comment]
Thanks @Manish-Giri 
I hope, that I'll continue the contribution on FCC projects :smile:

**Pull request merged**